Apps/Common/AppCommon.C

```diff
@@ -19,18 +19,22 @@
 
 bool SIM::AppXMLInputBase::parse (const TiXmlElement* elem)
 {
-  if (strcasecmp(elem->Value(),"geometry"))
+  if (!strcasecmp(elem->Value(),"geometry")) {
+    if (!utl::getAttribute(elem,"dimension",dim))
+      utl::getAttribute(elem,"dim",dim);
+  } else if (strcasecmp(elem->Value(),"postprocessing"))
     return true;
 
-  if (!utl::getAttribute(elem,"dimension",dim))
-    utl::getAttribute(elem,"dim",dim);
-
   std::string type;
   const TiXmlElement* child = elem->FirstChildElement();
   for (; child; child = child->NextSiblingElement())
-    if (!strcasecmp(child->Value(),"patchfile"))
+    if (!strcasecmp(child->Value(),"patchfile")) {
       if (utl::getAttribute(child,"type",type) && type == "lrspline")
         IFEM::getOptions().discretization = ASM::LRSpline;
+    } else if (!strcasecmp(child->Value(),"restartfile")) {
+      restartfile = utl::getValue(child, "restartfile");
+      utl::getAttribute(child, "level" , restartlevel);
+    }
 
```

Comment:
ja, pre-parsing blir litt misvisende. greia er at der har jeg en mekanisme for å parse en xml-tag, og samtidig legge til en ekvivalent kommandolinje-opsjon. alternativet er å legge den parsinga inn i en sim-klasse som var problematisk hvis dette ikke skulle gjøres på SIMoutput-nivå. jeg valgte derfor å bruke pre-parsing mekanismen. jeg kan flytte xml-tag'en hvor som helst, det føltes bare naturlig å putte den ved siden av den andre striden for min del.